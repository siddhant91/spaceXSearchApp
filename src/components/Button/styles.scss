@import '../../styles/main.scss';
.spacex-button-component {
	border-radius: $spacex-spacing-1;
	border: none;
	padding: $spacex-spacing-2;
	cursor: pointer;
	background-color: $spacex-color-secondary-green;
	outline: none;
	height: 35px;
	width: 100px;
	margin: $spacex-spacing-2 $spacex-spacing-3;
	font-size: $spacex-body-font-size-medium;
	@media only screen and (min-width: $spacex-breakpoint-mobile) and (max-width: $spacex-breakpoint-x-large) {
		height: $spacex-spacing-6;
		width: $spacex-spacing-8;
		font-size: $spacex-body-font-size;
		margin: 5px;
	}
	&.selected {
		background-color: $spacex-color-primary-green;
	}
}
