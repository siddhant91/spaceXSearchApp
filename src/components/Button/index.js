/**
 * Generic Button Component
 */
import { node, string, bool, func } from 'prop-types';
import clsx from 'clsx';

// Styles
import './styles.scss';

function Button({ additionalClasses, children, disabled, handleClick, isSelected, ...props }) {
	return (
		<button
			type="button"
			className={clsx('spacex-button-component', additionalClasses, {
				disabled,
				selected: isSelected,
			})}
			disabled={disabled}
			onClick={handleClick}
			data-testid="button"
			{...props}
		>
			{children}
		</button>
	);
}

Button.propTypes = {
	children: node.isRequired,
	disabled: bool,
	handleClick: func.isRequired,
	additionalClasses: string,
	isSelected: bool,
};

Button.defaultProps = {
	disabled: false,
	additionalClasses: '',
	isSelected: false,
};

export default Button;
