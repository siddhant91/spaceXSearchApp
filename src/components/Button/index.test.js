import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';

import Button from './index';

afterEach(cleanup);

const fn = jest.fn();
const renderComponent = ({ disabled = false, isSelected = false } = {}) => {
	return (
		<Button handleClick={fn} disabled={disabled} isSelected={isSelected}>
			Click Me
		</Button>
	);
};
it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent());
	expect(asFragment()).toMatchSnapshot();
});

it('Renders children props', () => {
	const { getByText } = render(renderComponent());
	expect(getByText('Click Me')).toBeInTheDocument();
});

it('Handles Button Click', async () => {
	const { getByTestId } = render(renderComponent());
	const button = getByTestId('button');
	fireEvent.click(button);
	await waitFor(() => {
		expect(fn).toHaveBeenCalledTimes(1);
	});
});

it('disables the Button', async () => {
	const { getByTestId } = render(renderComponent({ disabled: true }));
	const button = getByTestId('button');
	fireEvent.click(button);
	await waitFor(() => {
		expect(fn).toHaveBeenCalledTimes(0);
	});
});

it('the Button is selected', async () => {
	const { getByTestId } = render(renderComponent({ disabled: true, isSelected: true }));
	const button = getByTestId('button');
	expect(button.classList.contains('selected')).toBe(true);
});
