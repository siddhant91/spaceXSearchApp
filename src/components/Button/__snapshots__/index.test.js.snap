// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="spacex-button-component"
    data-testid="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
